contributor-book/src/guides/adding-a-new-operation-to-burn.md

```diff
@@ -0,0 +1,216 @@
+3. comparing the actual to expected output for lhs, rhs and regular operation
+
+generally, it seems preferable to use
+`actual_output_tensor.to_data().assert_approx_eq(&expected_tensor_data,3)` to `assert_eq!(...` due
+to occasional hiccups with floating point calculations.
+
+## Adding the Op to other backends
+
+most of these are fairly straightforward implementations. for reference here's pow's float
+implementation for torch, ndarray and candle backends:
+
+1. Torch implementation in
+   [burn-tch/src/ops/tensor.rs](https://github.com/tracel-ai/burn/blob/main/burn-tch/src/ops/tensor.rs#L461)
+   and the Op used in
+   [burn-tch/src/ops/base.rs](https://github.com/tracel-ai/burn/blob/4ca3e31601228952bb1c1492bc9cd2adf15b5cf1/burn-tch/src/ops/base.rs#L443)
+2. NdArray in
+   [burn-ndarray/src/ops/tensor.rs](https://github.com/tracel-ai/burn/blob/main/burn-ndarray/src/ops/tensor.rs#L443)
+3. Candle in
+   [burn-candle/src/ops/tensor.rs](https://github.com/tracel-ai/burn/blob/4ca3e31601228952bb1c1492bc9cd2adf15b5cf1/burn-candle/src/ops/tensor.rs#L481)
+
+This is where any calculation happens currently. Playing a guessing game with method names and
+seeing what completions are suggested will take you far. If you are having trouble figuring out how
+to do it from the docs for that backend,
+[try searching github for relevant function calls](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax).
+
+## Adding the Op to fusion and wgpu backends
+
+Adding an operator to these backends is fairly straightforward, though due to what these backends
+are for, involves a bit more indirection. Fusion, like autodiff, is not a target backend as much as
+a backend that enables certain functionality for other backends, in this case kernel fusion (which
+is currently only supported for `burn-wgpu`), so adding the operator won't involve doing any
+calculation, you'll just be describing how the generated code should look. Most of this can be
+copy/pasted/adjusted from other functions.
+
+here's how powf was added to burn fusion:
+
+1. added powf to the float ops under
+   [`burn-fusion/src/ops/float.rs`](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-fusion/src/ops/float.rs#L1758)
+2. added powf to the `FloatOperationDescription` enum under
+   [burn-fusion/src/stream/operation.rs](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-fusion/src/stream/operation.rs#L385)
+3. added powf to the implementations of `FloatOperationDescription` enum under
+   [burn-fusion/src/stream/context.rs](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-fusion/src/stream/context.rs#L726)
+
+Adding pow to wgpu was actually pretty easy due to the design. Element-wise tensor ops are just
+vectorized scalar ops, and given that raising a tensor to a scalar power prior to the tensor
+version, I just reused the code for scalar powf.
```

Comment:
This sentence is hard to understand. Can you rephrase it and remove the first person `I` ?